ifermi/plot.py

```diff
@@ -346,7 +351,10 @@ def _get_plot_data(
 
         isosurfaces = []
         if not hide_surface:
-            isosurfaces = self.fermi_surface.all_vertices_faces(spins=spin)
+            if plot_index is None:
+                isosurfaces = self.fermi_surface.all_vertices_faces(spins=spin)
+            else:
+                isosurfaces = self.fermi_surface.all_vertices_faces(spins=spin, band_index=plot_index)
```

Comment:
This can all be simplified to 
```suggestion
            isosurfaces = self.fermi_surface.all_vertices_faces(spins=spin, band_index=plot_index)
```